// nw-irc-ui/nw-irc-ui.js
irc = require('nw-irc')
crypto = require('crypto');

exports.ClientUI = ClientUI;

function ClientUI () {
    this.channel_id = 0;
    this.channel_panes = {};
}

ClientUI.prototype.getChannelPaneId = function (server, channel) {
    return crypto.createHash('md5').update(server + '\\' + channel).digest('hex');
}

ClientUI.prototype.createChannelPane = function (server, channel) {
    id = this.getChannelPaneId(server, channel);
    $('<li><a href="#' + id + '" data-toggle="tab">' + channel + '</a></li>').appendTo('#channel-nav');
    
    // fixme: might not work when a user and channel exist with the same name
    return $('<div class="tab-pane" id="' + id + '"></div>').appendTo('#channel-tabs')
}

ClientUI.prototype.getChannelPane = function (server, channel) {
    return $('#' + this.getChannelPaneId(server, channel));
}

ClientUI.prototype.initClient = function (client) {
    var self = this;
    var server = client.server;

    // set up event handlers
    client.on('message', function(dest, to, text, message) {
        if (dest === client.nickname && message.nick !== client.nickname) {
            self.createChannelPane(server, message.nick);
        }
    });

    client.on('message#', function(nick, to, text, message) {
        self.getChannelPane(server, to).append('<p>&lt;' + nick + '&gt;&nbsp;' + text + '</p>');
    });

    client.on('join', function(channel, nick, message) {
        if (nick == client.nickname) {
            self.createChannelPane(server, channel);
        }
    });
}
