// nw-irc/nw-irc.js
irc = require('irc')

exports.Client = Client;

Client = irc.Client
Client.prototype.joinChannel = function (channel) {
    this.join(channel)
}
Client.prototype.partChannel = function (channel) {
    this.part(channel)
}
Client.prototype.cyckeChannel = function (channel) {
    this.part(channel)
    this.join(channel)
}

function getChannelPane (server, channel) {
    return $('#' + server + '\\' + channel)
}

function createChannelPane (server, channel) {
    id = '' + server + '\\' + channel + '';
    $('<li><a href="#' + id + '" data-toggle="tab">' + channel + '</a></li>').appendTo('#channel-nav');
    
    // fixme: might not work when a user and channel exist with the same name
    return $('<div class="tab-pane" id="' + id + '"></div>').appendTo('#channel-tabs')
}

Client.prototype.init (server, args) {
    port = args.port ? args.port : 6667;
    ssl = args.ssl ? true : false;
    password = args.password ? args.password : '';

    c = new irc.Client(server, nickname, {
        'port': port,
        'secure': ssl,
        'password': password,
        'selfSigned': true
    });

    // set up event handlers
    c.on('message', function(dest, to, text, message) {
        if (dest === nickname) {
            createChannelPane(server, message.nick);
        }
    });

    c.on('message#', function(dest, to, text, message) {
        getChannelPane(server, dest).append('<p>' + message + '</p>');
    });

    c.on('join', function(channel, nick, message) {
        if (nick == nickname) {
            createChannelPane(server, channel);
        }
    });
}
